contracts/CitizenERC721.sol

```diff
@@ -89,6 +92,10 @@ contract CitizenERC721 is Initializable, ERC721Upgradeable, ERC721EnumerableUpgr
         override(ERC721Upgradeable, ERC721EnumerableUpgradeable)
     {
         super._beforeTokenTransfer(from, to, tokenId);
+
+        // TODO(@cameron): Is this in the right place?
```

Comment:
This looks right, but I would like to come up with a test case to ensure there is no way that this still executes in the event the transfer fails for some reason.